### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by ibr5500**
## in this [PR] we (@sarazaToussaint, and @ibr5500 ) have:

> - Solved the N+1 problem, when fetching all posts and their comments, and likes for a user.
> - Use Capybara to write integration tests for the following pages (User index page, User show page, Post index page, and Post show page).
> - Update the README file.

### CODE REVIEW

File: spec/features/post_spec.rb [hunk 1]
```diff
@@ -91,16 +91,15 @@
...
-      comment = @post1.comments.first
-      comment2 = @post1.comments.last
-      expect(page).to have_content(comment.text)
-      expect(page).to have_content(comment2.text)
+    scenario 'Should show the comment each comment user left ' do
+      @user = User.first
+      @post = Post.first
+      visit user_post_path(@user, @post)
+      expect(page).to have_text(@post.comments[0].text)
+      expect(page).to have_text(@post.comments[1].text)
```
Comment (dasileker) [inline spec/features/post_spec.rb]: - [ ] Hello and nice job on adding the tests but this test suit is failing due to the wrong argument provided, `hint: please you are trying to get a comment of a specific user in the post, not the post comment`

Capture d’écran 2022-08-30 145935

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @ibr5500,** 

### **Good job so far!**
There are some issues that you still need to work on to go to the next project but you are almost there!

## **Highlights**
- Linters are green ✅.
- PR is very nice 👌.
- The readme file is also good 👌.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ibr5500** [comment]
Thank you @dasileker for the code review.

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @ibr5500, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

 Please double-check the project link you have submitted

- [ ] This is the [project reqiurement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_devise.md)

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @ibr5500, This is @sja-thedude your code reviewer.

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

200w

 ## Highlights
✔️ Good job so far

 ## Required Changes ♻️
- [ ] Kindly make sure to fix one test case as it fails.
Screenshot 2022-08-30 at 5 58 15 PM

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ibr5500** [comment]
Thank @sja-thedude for the code review.

**Review by FrederickMih — APPROVED** [review]
Hi @ibr5500, @CollinsTatang , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- GitHub workflow has been followed :+1: 
- PR has a title and summary :+1: 
- Meaningful commit messages :+1: 
- All test cases are passing :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**